git/operations.go

```diff
@@ -1,12 +1,19 @@
 package git
 
+/*
+
+The purpose of this package is to provide a thin wrapper around the `git` CLI.
+
+At this layer of abstraction, it is generally inappropriate to add logic to this file that git would otherwise be aware of.  For example, whether or not a repo is readonly is not something you have to supply to `git push`.  If you attempt to `git push` on a readonly repo, git will return an error.  And so, carrying that same behavior to this module, if you try to use the `push` function on a readonly repo, that function should return an error.
```

Comment:
did you see the next sentence that goes on: `For example`?

The point of adding that was that you and I had made a decision that things that go beyond what arguments are available as flags to a regular `git` command are generally inappropriate to add in this file which is intended to be a light wrapper for git commands.

Since I wasted time (initially) writing such logic into this file only to realize later upon discussing with you that it was the wrong approach, I think it appropriate to write that design approach down so that no one in the future, perhaps, is tripped up by the same thing.  If this context were available to me when I started it would have been helpful and so I hope it will be helpful to the next person going forward.

Let me know what changes you request and I will happily comply.

Reply:
I wasn't questioning the motivation for the comment. I honestly have trouble interpreting it. Is it something like "these procedures will return an error if the operation is not allowed by the repo configuration"?

Perhaps the trouble is that it starts by explaining what the code doesn't do. If you turned that around and explained the design as it is, it would be easier to follow.

Reply:
I removed this from this PR to avoid the conversation at this time.